File: JS/02.reverseNumber.js
function reverseNumber(n) {
  let num = n;
  let reverse = 0;

  while (n != 0) {
    let x = n % 10;
    reverse = reverse * 10 + x;
    n = Math.floor(n / 10);
  }

  return reverse;
}

function recurReverseNumber(n, reverse) {
  if (n == 0) {
    return reverse;
  }
  rem = n % 10;
  reverse = reverse * 10 + rem;
  n = Math.floor(n / 10);

  return recurReverseNumber(n, reverse);
}

var ans = (n) => {
  return parseFloat(n.toString().split("").reverse().join(""));
};

var n = 1234;
var reverse = 0;
console.log(reverseNumber(n));
console.log(recurReverseNumber(n, reverse));
console.log(ans(n));

File: Python/number/02.reverseNumber.py
def reverseNumber(n):
    num = n
    reverse = 0
    
    while(num>0):
        rem = num%10
        reverse = reverse*10+rem
        num = num//10
        
    return reverse

def recurReverseNumber(n,reverse):
    if n == 0:
        return reverse
    rem = n%10
    reverse = reverse*10+rem
    n = n//10
    return recurReverseNumber(n,reverse)

n = 12345
reverse = 0
print(reverseNumber(n))
print(recurReverseNumber(n,reverse))
print(int(str(n)[::-1]))